src/legacy/core_plugins/ui_metric/public/services/telemetry_analytics.ts

```diff
@@ -39,28 +41,36 @@ export const createUiStatsReporter = (appName: string) => (
   }
 };
 
+export const trackUserAgent = (appName: string) => {
+  if (telemetryReporter) {
+    return telemetryReporter.reportUserAgent(appName);
+  }
+};
+
 interface AnalyicsReporterConfig {
   localStorage: any;
-  basePath: string;
   debug: boolean;
-  $http: ng.IHttpService;
+  kfetch: any;
 }
 
 export function createAnalyticsReporter(config: AnalyicsReporterConfig) {
-  const { localStorage, basePath, debug } = config;
+  const { localStorage, debug, kfetch } = config;
 
-  return createReporter({
+  return new Reporter({
     debug,
     storage: localStorage,
     async http(report) {
-      const url = `${basePath}/api/telemetry/report`;
-      await fetch(url, {
+      const response = await kfetch({
         method: 'POST',
-        headers: {
-          'kbn-xsrf': 'true',
-        },
-        body: JSON.stringify({ report }),
+        pathname: '/api/telemetry/report',
+        body: JSON.stringify(report),
+        headers: addSystemApiHeader({}),
       });
+
+      if (response.status !== 'ok') {
+        throw Error('Unable to store report.');
+      }
```

Comment:
Should we include the error from the response? Perhaps the code at least?
Otherwise we might just see failures without knowing why they fail.

Reply:
The `response` here is `200` as we dont want the browser to be showing errors if sending usage fails. the response here is always `200` with `status` `ok` or `fail`

Reply:
I don't follow... we _are_ throwing an error, right? So, shouldn't be clearer about why we threw an error?